pkg/apis/discovery/validation/validation_test.go

```diff
@@ -552,21 +610,99 @@ func TestValidateEndpointSliceUpdate(t *testing.T) {
 			},
 			expectedErrors: 0,
 		},
-		"deprecated address type": {
+		"node name set before + after, gate disabled": {
+			newEndpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{{
+					Addresses: []string{"10.1.2.3"},
+					NodeName:  utilpointer.StringPtr("foo"),
+				}},
+			},
+			oldEndpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{{
+					Addresses: []string{"10.1.2.3"},
+					NodeName:  utilpointer.StringPtr("foo"),
+				}},
+			},
+			expectedErrors: 0,
+		},
+		"node name set after, gate enabled": {
+			nodeNameGateEnabled: true,
+			newEndpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{{
+					Addresses: []string{"10.1.2.3"},
+				}},
+			},
+			oldEndpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{{
+					Addresses: []string{"10.1.2.3"},
+					NodeName:  utilpointer.StringPtr("foo"),
+				}},
+			},
+			expectedErrors: 0,
+		},
+
+		// expected errors
+		"invalide node name set after, gate enabled": {
+			nodeNameGateEnabled: true,
+			newEndpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{{
+					Addresses: []string{"10.1.2.3"},
+					NodeName:  utilpointer.StringPtr("INVALID foo"),
+				}},
+			},
+			oldEndpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{{
+					Addresses: []string{"10.1.2.3"},
+				}},
+			},
+			expectedErrors: 1,
+		},
+		"node name set after, gate disabled": {
```

Comment:
This test name should be "node name set before, gate disabled"

Reply:
Good catch, updated this to actually do what the name was describing.